redfish-core/lib/rde.hpp

```diff
@@ -50,32 +56,124 @@ struct Resource
+std::string handleDeferredBindings(const std::string& bejJsonInput)
 {
-    BMCWEB_LOG_DEBUG("RDE : Get handler.");
-    if (!redfish::setUpRedfishRoute(app, req, asyncResp))
+    // TDDO implement handleDeferredBindings
+    return bejJsonInput;
+}
+
+inline bool rdeHandleCreateTask(const boost::system::error_code& ec2,
+                                sdbusplus::message_t& msg,
+                                const std::shared_ptr<task::TaskData>& taskData)
+{
+    if (ec2)
     {
-        return;
+        return task::completed;
     }
 
-    // TODO: Implement the GET request handling logic here.
-    asyncResp->res.result(boost::beast::http::status::ok);
-    asyncResp->res.jsonValue["status"] = "success";
-}
+    dbus::utility::DBusPropertiesMap changedProperties;
+    msg.read(changedProperties);
 
-// PATCH handler
-inline void rdePatchHandler(App& app, const crow::Request& req,
-                            const std::shared_ptr<bmcweb::AsyncResp>& asyncResp)
-{
-    BMCWEB_LOG_DEBUG("RDE : Patch handler.");
+    const std::string* data = nullptr;
+    const uint16_t* rc = nullptr;
+    nlohmann::json taskEntry;
 
-    if (!redfish::setUpRedfishRoute(app, req, asyncResp))
+    for (const auto& propertyPair : changedProperties)
     {
-        return;
+        if (propertyPair.first == "payload")
+        {
+            data = std::get_if<std::string>(&propertyPair.second);
+
+            const auto& updatedData = handleDeferredBindings(*data);
+            taskEntry["Payload"] = updatedData;
+        }
+        else if (propertyPair.first == "return_code")
+        {
+            rc = std::get_if<uint16_t>(&propertyPair.second);
+            taskEntry["ReturnCode"] = *rc;
+        }
+    }
+    if (data == nullptr || rc == nullptr)
+    {
+        BMCWEB_LOG_ERROR("Invalid property or property not found");
+        taskData->messages.emplace_back(messages::internalError());
+        return task::completed;
     }
 
+    std::string index = std::to_string(taskData->index);
+    taskData->messages.emplace_back(taskEntry);
+    taskData->messages.emplace_back(messages::taskCompletedOK(index));
+    taskData->state = "Completed";
+
+    return task::completed;
+}
+
+inline void rdeCreateTask(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                          task::Payload&& payload,
+                          const sdbusplus::message::object_path& objPath)
+{
+    std::string matchString =
+        ("type='signal',"
+         "path='" +
+         objPath.str +
+         "',"
+         "interface='xyz.openbmc_project.RDE.OperationTask',"
+         "member='TaskUpdated'");
+
+    std::shared_ptr<task::TaskData> task = task::TaskData::createTask(
+        std::bind_front(rdeHandleCreateTask), matchString);
+    task->startTimer(std::chrono::minutes(5));
+    task->populateResp(asyncResp->res);
+    task->payload.emplace(std::move(payload));
+}
+
+// TODO: Make OperationIDs persistent or something similar to avoid operation id
```

Comment:
As i mentioend commented earlier , any BMCweb reboot will stop taking request . either use persist the operation id or look how task manger handles this.